### ISSUE TITLE
Attaching `TemplatePortal` via `DomPortalHost` always returns an empty `Map`

### ISSUE THREAD

**Issue by emilio-martinez** [angular/material2#3858]
#### Bug, feature request, or proposal:
Feature request

#### What is the expected behavior?
Setting a `TemplatePortal`'s locals should get passed to the view when attaching via `DomPortalHost`, and then after attaching, the locals should return all the locals in view.

https://github.com/angular/material2/blob/master/src/lib/core/portal/dom-portal-host.ts#L81

#### What is the current behavior?
Setting a `TemplatePortal`'s locals doesn't get passed to the view when creating an EmbeddedView, i.e., https://github.com/angular/material2/blob/master/src/lib/core/portal/dom-portal-host.ts#L66. Furthermore, an empty map gets returned after attaching, i.e., https://github.com/angular/material2/blob/master/src/lib/core/portal/dom-portal-host.ts#L81.

Of course, I see the comment that this is a pending task; however, when not passing the context param, the context returns as an `Object`, not a map as the `Portal` API describes, which I guess makes this relatively non-trivial since setting a Map as the context doesn't seem to interop with the view.

#### What is the use-case or motivation for changing an existing behavior?
Full-circle with the API described. as mentioned above, I realize this is a pending task, but thought I might surface it since I'm currently struggling with it on my project.

#### Which versions of Angular, Material, OS, browsers are affected?
Angular 2.4.10, Material 2.0.0-beta.2.

**Comment by jelbourn** [comment]
Just curious- what do you need the locals for? I intended to implement this (as you saw from the comment), but then put it on the backburner because I didn't have any good use cases for why anyone would need it.

**Comment by emilio-martinez** [comment]
I caveat my response by saying that—to be honest—I'm having a bit of a hard time remembering _exactly_ what my use case was at the time (I worked around it for now); however, I'll do my best:

I have a `TemplateRef` that I wanted to use as a "factory" for creating `TemplatePortal`s programmatically and attaching them to overlays; however, my template has locals. The disconnect moment for me came at the time of trying to pass values as locals to the template. The way I saw it I had two options:
* I could set the `locals` on my newly created `TemplatePortal`, but on `attachTemplatePortal` in the `DomPortalHost` the context isn't passed to `createEmbeddedView` (https://github.com/angular/material2/blob/master/src/lib/core/portal/dom-portal-host.ts#L66).
* I could get the map of `locals` after `attachTemplatePortal` happens, and then set my values on the `Map` then, but obviously that's still returning a new `Map`.

Additionally, doing some tests on my end, I only got my locals reflected in the View when passing an `Object` literal to `createEmbeddedView`, not a `Map`. Taking the values from the `Portal` `locals` and copying them into an object to pass would work just fine; however, the API would be disjointed because `createEmbeddedView` would need to return a `Map`; which is where this situation stopped for me.

I know other people on my team have bumped into the issue of not getting the locals back when using `TemplatePortal`s for other use cases, but it's mostly been an inconvenience; we've simply figured out ways to work around it. It may also be worth mentioning that we've been using those modules more as "Component Dev Kit" pieces rather than via the components themselves.

Hope this helps!
